cmd/hope/utils/utils.go

```diff
@@ -51,3 +55,47 @@ func ReplaceParametersWithSubstitutor(t *hope.TextSubstitutor, parameters []stri
 
 	return string(*t.Bytes), nil
 }
+
+func ReplaceParametersInDirectory(dir string, parameters []string) error {
+	return filepath.Walk(dir, func(apath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		str, err := ReplaceParametersInFile(apath, parameters)
+		if err != nil {
+			return err
+		}
+
+		return fileutil.WriteFile(str, apath)
+	})
+}
+
+// ReplaceParametersInDirectoryCopy - Copy the provided directory, and replace
+//   parameters in the files.
+// Returns the temp path to the copied directory, and the caller must clean ip
+//   that directory itself.
+func ReplaceParametersInDirectoryCopy(dir string, parameters []string) (string, error) {
+	tempDir, err := ioutil.TempDir("", "*")
+	if err != nil {
+		return "", err
+	}
+
+	if err := fileutil.CopyDirectory(dir, tempDir); err != nil {
```

Comment:
Pretty lazy. This could probably be updated to map files as it copies them over, but probably not worth it today.